rct229/rulesets/ashrae9012019/section18/section18rule1.py

```diff
@@ -0,0 +1,206 @@
+
+            return is_undetermined
+
+        def get_manual_check_required_msg(self, context, calc_vals=None, data=None):
+            zone_b = context.baseline
+            zone_id_b = zone_b["id"]
+
+            hvac_systems_serving_zone_b = data["hvac_systems_serving_zone_b"][zone_id_b]
+            expected_system_type_b = hvac_systems_serving_zone_b[
+                "expected_system_type_b"
+            ]
+
+            if hvac_systems_serving_zone_b["system_origin"] == SYSTEMORIGIN.G311D:
+                advisory_note = (
+                    f"HVAC system type {expected_system_type_b} was selected for this zone based on ASHRAE 90.1 Appendix G3.1.1.d, which reads: "
+                    f"For laboratory spaces in a building having a total laboratory exhaust rate greater than 15,000 cfm, use a single system of type 5 or 7 serving only those spaces. "
+                    f"We have determined that laboratory spaces in this building have a total exhaust greater than 15,000 cfm, but we cannot determine with certainty that at least 15,000cfm of the exhaust is dedicated to laboratory purposes. "
+                    f"If the building laboratory exhaust is greater than 15,000cfm, the system type for this zone should be {expected_system_type_b}."
+                )
+
+            elif hvac_systems_serving_zone_b["system_origin"] == SYSTEMORIGIN.G311E:
+                advisory_note = (
+                    f"HVAC system type {expected_system_type_b} was selected for this zone based on ASHRAE 90.1 Appendix G.3.1.1.e which reads: "
+                    f"Thermal zones designed with heating-only systems in the proposed design serving storage rooms, stairwells, vestibules, electrical/mechanical rooms, "
+                    f"and restrooms not exhausting or transferring air from mechanically cooled thermal zones in the proposed design shall use system type 9 or 10 in the baseline building design.` "
+                    f"We expect that this space is a vestibule, but cannot make a determination with 100% accuracy. If the zone is one of the listed space types, then the system type should be {expected_system_type_b}."
+                )
+
+            else:
+                advisory_note = ""
+
+            return advisory_note
+
+        def rule_check(self, context, calc_vals=None, data=None):
+            is_system_part_of_expected_sys_type_b = calc_vals[
+                "is_system_part_of_expected_sys_type_b"
+            ]
+
+            return is_system_part_of_expected_sys_type_b
+
+        def get_fail_msg(self, context, calc_vals=None, data=None):
+            zone_b = context.baseline
+            zone_id_b = zone_b["id"]
+
+            hvac_systems_serving_zone_b = data["hvac_systems_serving_zone_b"][zone_id_b]
+            system_type_origin_b = hvac_systems_serving_zone_b["system_origin"]
+            expected_system_type_b = hvac_systems_serving_zone_b[
+                "expected_system_type_b"
+            ]
```

Comment:
Similarly, it is likely you used the wrong data
The `hvac_systems_serving_zone_b` should really be `target_baseline_system_dict`

Reply:
Addressed.